### ISSUE TITLE
Explicit Music should have [E] instead of complete "explicit"

### ISSUE THREAD

**Issue by RSickenberg** [lowlighter/metrics#1674]
### Discussed in https://github.com/lowlighter/metrics/discussions/1673

<div type='discussions-op-text'>

<sup>Originally posted by **RSickenberg** December  7, 2024</sup>
<img width="238" alt="image" src="https://github.com/user-attachments/assets/52fd587d-1cf1-482e-b248-56d61c620b08">
Here is what is rendered on my side.</div>

**Comment by RivenSkaye** [comment]
I'd suggest doing what Spotify does here, and adding it behind the artist name, as either the regional indicator E or, keeping it plaintext, `[E]`

**Comment by RSickenberg** [comment]
> I'd suggest doing what Spotify does here, and adding it behind the artist name, as either the regional indicator E or, keeping it plaintext, `[E]`

Everything except plain text.

**Comment by yuanweize** [comment]
Hi @RSickenberg — I am maintaining an independent community fork of lowlighter/metrics focused on compatibility, security fixes, and ongoing maintenance:
https://github.com/yuanweize/metrics-community

I have mirrored and am tracking this issue in the community fork tracker here:
https://github.com/yuanweize/metrics-community/issues/31

The original report and attribution remain credited to this upstream issue. The fork is independent and is not an official successor to lowlighter/metrics.
If this problem still affects your workflows, testing or further feedback on the maintained fork would be very welcome.

**Comment by RSickenberg** [comment]
@yuanweize Have you fixed the codding habits ?

**Comment by yuanweize** [comment]
Yes! The coding habits plugin (`plugin_habits`) issues were primarily caused by changes in GitHub's Events API (PushEvent payloads omitting `commits` arrays or returning sparse/null commit entries and null authors, which caused `TypeError: Cannot read properties of undefined (reading 'flatMap')` and author destructuring crashes).

In `yuanweize/metrics-community`, we have consolidated and hardened the habits plugin:
- Guarded `payload.commits` against undefined/null arrays and sparse commit elements
- Added null-safe author checking (`author?.login`, `author?.email`, `author?.name`)
- Added null-safe patch and timezone offset handling
- Verified with regression test suite (`tests/plugins/events-api.test.js`)

It is included in release [`v3.35.0-community.1`](https://github.com/yuanweize/metrics-community/releases/tag/v3.35.0-community.1) and on `main`. If you'd like to test your configuration on the community fork, feel free to give it a spin:
```yaml
uses: yuanweize/metrics-community@main
```
